cmd/zetae2e/local/utils.go

```diff
@@ -100,3 +103,52 @@ func waitKeygenHeight(
 		logger.Info("Last ZetaHeight: %d", response.Height)
 	}
 }
+
+func MonitorTxPriorityInBlocks(ctx context.Context, conf config.Config, errCh chan error) {
+	rpc, err := rpchttp.New(conf.RPCs.ZetaCoreRPC, "/websocket")
+	if err != nil {
+		errCh <- err
+	}
+
+	ticker := time.NewTicker(2 * time.Second)
+	defer ticker.Stop()
+	for {
+		select {
+		case <-ctx.Done():
+			errCh <- nil
+		case <-ticker.C:
+			block, err := rpc.Block(ctx, nil)
+			if err != nil {
+				errCh <- err
+			}
+
+			// iterate txs events and check if some MsgEthereumTx is included above crosschain and observer txs
+			nonSystemTxFound := false
+			for _, tx := range block.Block.Txs {
+				txRes, err := rpc.Tx(context.Background(), tx.Hash(), false)
+				if err != nil {
+					continue
+				}
+
+				for _, ev := range txRes.TxResult.Events {
+					for _, attr := range ev.Attributes {
+						if attr.Key == "msg_type_url" {
+							if strings.Contains(attr.Value, "zetachain.zetacore.crosschain.MsgVote") ||
+								strings.Contains(attr.Value, "zetachain.zetacore.observer.MsgVote") ||
+								strings.Contains(attr.Value, "zetachain.zetacore.observer.MsgAddBlameVote") {
```

Comment:
I don't think we should rely on `strings.Contains(attr.Value, "zetachain.zetacore.crosschain.MsgVote")`
We should do a equal check with a complete message name.

We can create a method for checking the URL is a system tx directly, maybe we already have it for the logic in the application mempool